files/default/bin/deploy.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+PUBLIC=0
+
+run_help() {
+  echo
+  tput bold
+  echo " Name:"
+  tput sgr0
+  echo "    $0 - deploy target image to OpenStack for further testing."
+  echo
+  tput bold
+  echo " Usage:"
+  tput sgr0
+  echo "    $0 [-p] -f [FILE] -n [IMG_NAME] -r [PR_NUM]"
+  echo
+  tput bold
+  echo " Options:"
+  tput sgr0
+  echo "    -p          - Publish (publicly) the target image ('\$PR'), renaming the"
+  echo "                  old \$IMG_NAME to \$PR_NUM and making that image private"
+  echo
+  tput bold
+  echo " Arguments:"
+  tput sgr0
+  echo
+  echo "    FILE      - Fully built image to be uploaded to OpenStack if"
+  echo "                [-p] is not specified."
+  echo
+  echo "    IMG_NAME  - Name of the image to replace if [-p] is specified."
+  echo
+  echo "    PR_NUM    - GitHub Pull Request number to name private images."
+  echo
+  exit 0
+}
+
+while getopts "hpf:n:r:" opt ; do
+  case $opt in
+    h)
+      run_help
+      ;;
+    p)
+      PUBLIC=1
+      ;;
+    f)
+      FILE="$OPTARG"
+      ;;
+    n)
+      IMG_NAME="$OPTARG"
+      ;;
+    r)
+      PR_NUM="$OPTARG"
+      ;;
+  esac
+done
+
+set -xe
+[ -z "$PR_NUM" ] && echo "Error: PR_NUM not set. Try '$0 -h'" && exit 1
+
+if [ $PUBLIC == 0 ]
+  then
+      [ -z $FILE ] && echo "Error: FILE not set. Try '$0 -h'" && exit 1
+      openstack image create --file $FILE --disk-format qcow2 "$PR_NUM"
+      exit 0
+else
+  [ -z "$IMG_NAME" ] && echo "Error: IMG_NAME not set. Try '$0 -h'" && exit 1
+  openstack image set --name "$IMG_NAME" --private "$PR_NUM"
```

Comment:
The name would be `$IMG_NAME - $PR_NUM` because we might change multiple JSONS in one PR